cowprotocol/cowswap
# feat(smart-contract-orders): hide presign order until wallet returns tx for Safe app limit bundling tx

# Summary

Hide order from UI until it's confirmed by the wallet

https://github.com/cowprotocol/cowswap/assets/43217/78224eed-8042-47b8-8df0-56a00a29a325

If user rejects the tx or it's never signed, the order will remain hidden in the same device.
It'll still be visible in different devices and on the Explorer

## ⚠️ Notes ⚠️
- For now the **only** path implemented is for Safe wallet, loaded as an App, for Limit orders, bundling approve + presign
- Will implement remaining paths/wallet in a follow up PR
- This is an experimental PR and I'm focusing on only one flow as the major changes in the state/middleware

  # To Test

1. Load app inside the Safe and open the limit orders
2. Pick a pair of which the sell token needs approval
3. Confirm the order until the Safe confirmation is displayed
* By this point, the order is already created on the backend and normally you would see the pop up and hear the sound
* There should be no pop up and no sound
4. Reject the tx
* Before this change, you would see the order in the list in a pre-sign stage
* The new order should not be visible anywhere
5. Open the Explorer and check the orders for you account
* The order should be shown there as presign pending state
6. Repeat step `3`, but this time submit/execute the tx
* The order should now appear is presign state
* There should be a pop up
* The moo sound should play
* Once executed/mined, the order should change to open and follow the regular flow

Please also test the EOA and Ethflow notifications and sounds, as this change touches on all of these areas.
Everything should behave the same as before.
